### PR TITLE
Add postgres schema to the search_path on database connection

### LINKED ISSUE #12505: Postgres schema support
- Gitea version (or commit ref): 1.12.3
- Git version:
- Operating system: CentOS 8
- Database (use `[x]`):
  - [x ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x ] Not relevant
- Log gist:
2020/08/17 06:59:17 ...m.io/xorm/core/db.go:154:QueryContext() [I] [SQL] SELECT user_id, count(*) AS count FROM notification WHERE user_id IN (SELECT user_id FROM notification WHERE updated_unix >= $1 AND updated_unix < $2) AND status = $3 GROUP BY user_id [1597661945 1597661955 1] - 260.505µs
2020/08/17 06:59:17 ...ource/manager_run.go:41:Run() [E] Unable to get UIDcounts: pq: relation "notification" does not exist

## Description

I'm using Postgres (11) and I want to use a schema (-> I don't want my data in a public schema in my cluster and I may want to have dev and test and prod instances)
I created a schema "prod" and configured it. After generating the schema, which seems successful, I soon get a "500" in the UI.
I look into the log and see this:
...
2020/08/17 06:59:17 ...m.io/xorm/core/db.go:154:QueryContext() [I] [SQL] SELECT user_id, count(*) AS count FROM notification WHERE user_id IN (SELECT user_id FROM notification WHERE updated_unix >= $1 AND updated_unix < $2) AND status = $3 GROUP BY user_id [1597661945 1597661955 1] - 260.505µs
2020/08/17 06:59:17 ...ource/manager_run.go:41:Run() [E] Unable to get UIDcounts: pq: relation "notification" does not exist
...

Looking at the sql that was generated, I have to agree - there is not table "notification": that is because there only is a prod.notification. 

I assume the problem is you do not qualify the table names properly for schema support?

### PR DESCRIPTION
**Pull request by zeripath**
Rather than rely on the user running the gitea server and db setting the
schema search_path correctly - if Gitea is run with a schema we should
simply set the search_path to have that schema first in the path.

Fix #12505

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

File: models/models.go [hunk 1]
```diff
@@ -153,6 +153,16 @@ func getEngine() (*xorm.Engine, error) {
 		engine.Dialect().SetParams(map[string]string{"DEFAULT_VARCHAR": "nvarchar"})
 	}
 	engine.SetSchema(setting.Database.Schema)
+	if setting.Database.UsePostgreSQL && len(setting.Database.Schema) > 0 {
+		// Add the schema to the search path
+		if _, err := engine.Exec(`SELECT set_config(
+			'search_path',
+			? || ',' || current_setting('search_path'),
```
Comment (silverwind) [inline models/models.go:162]: Not familar with this syntax. Can you explain what this line does?

Comment (ptman) [reply models/models.go:162]: `||` is (postgre?)sql for string concatenation

Comment (silverwind) [reply models/models.go:162]: SQL sure has weird operators. Maybe `CONCAT()` would be more clear?

Comment (zeripath) [reply models/models.go:162]: || is standard ANSI SQL.

Comment (silverwind) [reply models/models.go:162]: Fine, whatever 😉

**Review by silverwind — APPROVED** [review]
Maybe it'd be wise to quote the schema name in case it contains uppercase characters, but other than that lgtm.

**Review by lafriks — APPROVED** [review]

**Comment by lafriks** [comment]
🚀

**Pull request merged**